src/validation.cpp

```diff
@@ -4889,14 +4889,20 @@ bool Chainstate::ReplayBlocks()
     }
 
     // Rollback along the old branch.
+    int nForkHeight{pindexFork ? pindexFork->nHeight : 0};
+    if (pindexOld && pindexOld != pindexFork) {
+        LogInfo("Rolling back from %s (%i to %i)", pindexOld->GetBlockHash().ToString(), pindexOld->nHeight, nForkHeight);
+    }
     while (pindexOld != pindexFork) {
         if (pindexOld->nHeight > 0) { // Never disconnect the genesis block.
             CBlock block;
             if (!m_blockman.ReadBlock(block, *pindexOld)) {
                 LogError("RollbackBlock(): ReadBlock() failed at %d, hash=%s\n", pindexOld->nHeight, pindexOld->GetBlockHash().ToString());
                 return false;
             }
-            LogInfo("Rolling back %s (%i)", pindexOld->GetBlockHash().ToString(), pindexOld->nHeight);
+            if (pindexOld->nHeight % 10'000 == 0) {
+                LogInfo("Rolling back %s (%i)", pindexOld->GetBlockHash().ToString(), pindexOld->nHeight);
+            }
```

Comment:
Previously every block would have been printed. Now the start will be printed and then every 10'000th, but not the end, e.g.:

```
rolling back from 54321 to 29238
rolling back 50000
rolling back 40000
rolling back 30000
// and nothing more
```
Would it be useful to indicate that the process has completed? E.g. one more message "rolled back successfully to 29238" after the loop?

Same for the "rolling forward" case.